app/helpers/katello/katello_urls_helper.rb

```diff
@@ -37,11 +37,73 @@ def subscription_manager_configuration_url(host = nil, rpm = true, hostname: nil
+          fail MultiEnvironmentNotSupportedError,
+            "Host #{@host.name} must be subscribed to only a single content view & environment or subscribe to the default organization content view for liveimg provisioning."
+        end
+        content_path = [@host.single_content_view.label, content_path].join('/')
+      end
+
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_content_path(
+        @host.single_lifecycle_environment, content_path)
+      "#{url}/pulp/content/#{path}"
+    end
+    def repository_url(content_path, content_type = nil, schema = 'http')
+      return content_path if content_path =~ %r|^([\w\-\+]+)://|
+      url = if @host.content_source
+              "#{schema}://#{@host.content_source.hostname}"
+            else
+              foreman_settings_url(schema)
+            end
+      content_path = content_path.sub(%r|^/|, '')
+
+      # If the host is consuming from the default CV, use the library repository path provided in content_path.
+      # If the host has multiple environments, try to find the file among all the repositories.
+      host_cvs = @host.content_views
+      file_repo = nil
+      lifecycle_environment = nil
+      unless host_cvs.include?(::Katello::ContentView.default)
+        if content_type != 'isos'
+          fail ::Katello::Errors::InvalidRepositoryTypeError, 'Only file repositories (isos) may be used for the content_type when the host is not subscribed to the Default Organization View.'
+        end
+        potential_repos = ::Katello::Repository.in_content_views(host_cvs).where(environment_id: @host.lifecycle_environments.pluck(:id))
+        # Find the repositories that the file unit is in
+        file_repos = potential_repos.joins(:files).where(files: {path: ::File.basename(content_path)})
+        # More than one archived repo means the file exists in another, separate repository that the host could consume from.
+        if file_repos.archived.count > 1
+          fail ::Katello::Errors::ConflictException, "File #{content_path} found in multiple repositories: #{file_repos.pluck(:name)}. Ensure only one File repository contains the liveimg."
+        elsif file_repos.empty?
+          fail ::Katello::Errors::NotFound, "File #{content_path} not found in any of the assigned repositories."
+        else
+          # The user cannot specify the LCE to use, so use any of them that has the file in question.
+          file_repo = file_repos.sort_by(&:pulp_id).first
+          lifecycle_environment = file_repo.first.environement
+          content_path = [file_repo.first.content_view.label, content_path].join('/')
+        end
+      else
+        # If the host is subscribed to the default org view, use Library.
+        lifecycle_environment = ::Katello::KTEnvironment.library.where(organization_id: @host.organization.id)
+      end
+
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_content_path(
+        lifecycle_environment, content_path)
```

Comment:
Can we move this out of glue while we're touching this and put this in repository model layer?